straxen/plugins/acquisition_monitor.py

```diff
@@ -0,0 +1,109 @@
+import strax
+import straxen
+import numba
+import numpy as np
+from numba import njit
+from frozendict import frozendict
+
+export, __all__ = strax.exporter()
+
+@strax.takes_config(
+    # V1495 busy module:
+    # Generates a 25 ns NIM pulse whenever a veto begins and a 25 ns NIM signal when it ends. 
+    # A new start signal can occur only after the previous busy instance ended.
+    # 1ms (1e6 ns) - minimum busy veto length, or until the board clears its memory
+    
+    # DDC10 High Energy Veto:
+    # 10ms (1e7 ns) - fixed HE veto length in XENON1T DDC10, 
+    # in XENONnT it will be calibrated based length of large S2 SE tails
+    
+    strax.Option('max_veto_gap', default = int(5e8),
+                 help='Maximum separation between veto stop and start pulses [ns]'),
+    strax.Option('channel_map', track=False, type=frozendict, 
+                 help="frozendict mapping subdetector to (min, max)"
+                      "channel number."))
+
+@export
+class VetoIntervals(strax.OverlapWindowPlugin):
+    """ Merge start and stop times in acquisition monitor data 
+    for the low energy and high energy channels, as well as for the high energy veto 
+    no tag  <= V1495 busy veto for tpc channels
+    bb tag  <= V1495 busy veto for high energy tpc channels
+    hev tag <= DDC10 hardware high energy veto
+    """
+        
+    __version__ ='0.0.2'
+    depends_on = ('raw_records_aqmon') 
+    provides  = ('veto_intervals',)
+    data_kind = ('veto_intervals',)
+
+    aqmon_channel_names = ('sum_wf','m_veto_sync','hev_stop', 'hev_start', 'bb_stop', 'bb_start','stop', 'start')
+    duration_dtype = [(('Interval between start and stop in [ns]', 'interval'), np.int64)]
+    
+    # Using the time & endtime dtypes for busy so strax does not complain about lack of endtime and time dtypes
+    def infer_dtype(self):
+        dtype = []
+        for veto in ('','hev_','bb_'):
+            dtype += [(('Start '+ veto +'veto time since unix epoch [ns]', veto+'time'), np.int64)]
+            dtype += [(('Stop '+ veto +'veto time since unix epoch [ns]', veto+'endtime'), np.int64)]
+            dtype += [(('Duration of '+ veto +'veto time since unix epoch [ns]', veto+'interval'), np.int64)]
```

Comment:
Maybe an idea, as this is the difference between the two above, could we use one of the three to avoid saving things more than once?

Reply:
Sorry I don't understand. What do you mean by using one of the three?

Reply:
stop - start = interval, hence you wouldn't need to save start, stop and interval. But on the other hand, it's not going to be huge amounts of data stored like this right so it's not a big issue.

Reply:
I thought it would be just easier to put this later into mongo while already having the interval varible, while we can use the start/stop times themselves for a veto proximity to event plugin, etc. Obviously if its getting too big these inerval dtypes can get removed.